lib/DateTime.vala

```diff
@@ -56,6 +56,37 @@ namespace Granite.DateTime {
     }
 
     /**
+     * Compares a {@link GLib.DateTime} to {@link GLib.DateTime.now_local} and @return a localized, relative date and time string
+     */
+    public string get_relative_datetime (GLib.DateTime date_time) {
```

Comment:
* The function should be `static`.
* It would be better if you could pass a different `now` so you could calculate between different spans like with default being to `now_local`:

```
public static string get_relative_datetime (GLib.DateTime date_time, GLib.DateTime now = new GLib.DateTime.now_local ()) {
...
}
```

* You should make the documentation more extensive on what the function actually does, not just what it returns like: that it can return "Yesterday", "x h ago" etc.
* You should declare all the parameters and returns like this:
```
/**
 * ...
 * @param date_time the {@link GLib.DateTime} that represents the past point in time
 * @param now the {@link GLib.DateTime} that represents the current time, by default {@link GLib.DateTime.now_local}.
 * @return a localized, relative date and time string [possibly more details here]
 * 
```

Reply:
Do you have a use case for a different frame of reference? This would complicate things significantly, none of the existing strings would make since without the frame of reference being "now".

Reply:
It's not really about "use case" here. You can't just assume that the right "now" moment will always be `DateTime.now_local`. There can be possibility that the API user will want to pass e.g: `now_utc`, `now` or even a custom DateTime that was constructed manually with getting the date from external sources like the Internet.

Reply:
We shouldn't make this overly complicated for a feature that will never be used. Why do you think we can't assume that now local is what is wanted? Afaik that is what all of the existing examples are requesting. I think we need a solid example that not using now local is a valid desire before increasing complexity.

Reply:
It's would not be overly complicated. You set the default and that's the default. The user can call the API just as he could do this now, but if he wants to, he can provide an additional argument to set that.